### ISSUE TITLE
chore(infra): automate setup process

### ISSUE THREAD

**Issue by JowiAoun** [cuhacking/2025#155]
## 🆘 **What is the problem at hand?**

Contributors have to setup their development environment to start contributing. The setup process found in our [installation documentation](https://docs.cuhacking.ca/installation) (as of 2024-12-02) has many steps, and could be reduced down. We could automate setting up development environments, and potentially environment secrets.

- The development environment can contain features that make development easier for this specific project.
- For secrets, automating would mean creating and retrieving secrets with only a few commands.

This issue should implement the most appropriate way to proceed with the above.

## 🚀 **What is the impact of the problem? How is this problem important? How does it affect users/developers?**

We have had issues where Node.js is installed a different way on one contributor's system, causing builds to fail on their environment. That is why we need a setup process, preferably with Docker & [dev containers](https://code.visualstudio.com/docs/devcontainers/containers).

The setup process has many steps, and is largely dependent on the contributor's development system. That's why setting up the development environment for contributors will greatly reduce the likelihood for issues to be system related.

Additionally, requirements for environment variables may grow as we expand on services used, therefore more time from contributors will be spent setting up & updating environment variables. Automating the process of setting up these variables also reduces user error.

**Comment by JeremyFriesenGitHub** [comment]
would this kinda get refactored into creating containers/dev environment containers?

**Comment by JowiAoun** [comment]
> would this kinda get refactored into creating containers/dev environment containers?

Yes, thanks for mentioning it. I have updated the description to reflect this change.

**Comment by HasithDeAlwis** [comment]
**Closing because we used Nix to automate this process**
